apps/dapp/src/components/Pages/Ascend/components/LBPForm/use-pool-factory.ts

```diff
@@ -0,0 +1,91 @@
+import { BigNumber, Contract } from 'ethers';
+
+import liquidityBootstrappingPoolAbi from 'data/abis/liquidityBootstrappingPool.json';
+
+import { useWallet } from 'providers/WalletProvider';
+import { useEffect, useState } from 'react';
+import { formatEther, parseEther } from 'ethers/lib/utils';
+
+export interface CreatePoolParams {
+  name: string;
+  symbol: string;
+  tokenAddresses: string[];
+  weights: BigNumber[];
+  feePercentage: number;
+  swapEnabledOnStart: boolean;
+}
+
+export const useFactoryContract = () => {
+  const { signer } = useWallet();
+
+  const [ownerAddress, setOwnerAddress] = useState<string>();
+
+  useEffect(() => {
+    if (!signer) {
+      return;
+    }
+
+    (async () => {
+      const ownerAddress = await signer.getAddress();
```

Comment:
should be able to just do `const { wallet } = useWallet()`  to get the user address

Reply:
https://github.com/TempleDAO/temple/pull/461/commits/d6d1fcbad30026cafa8551085e227526b99551e5